phc-sf-spec.md

```diff
@@ -21,11 +21,12 @@ string that differs from the format herein described.
 
 We define the following format:
 
-       $<id>[$<param>=<value>(,<param>=<value>)*][$<salt>[$<hash>]]
+       $<id>[$v=<version>][$<param>=<value>(,<param>=<value>)*][$<salt>[$<hash>]]
```

Comment:
There's a somewhat ambiguous case:

```
$<id>[$v=<version>](,<param>=<value>)+][$<salt>[$<hash>]]
```

It'd be good to clarify what behavior parsers have for this case:

- MUST NOT interpret the `v` parameter as a version
- MUST interpret the `v` parameter as a version

Since a separated `[$v=<version>]$...` parameter is already in the wild, I would suggest MUST NOT here.

Reply:
I don't know if it helps, but you can refer to this implementation:
https://github.com/simonepri/phc-format/blob/master/index.js#L136-L224

If I remember correctly, in the case you mentioned my parser does not interpret `v` as the version but as a parameter.

> Since a separated [$v=<version>]$... parameter is already in the wild

What are you referring to exactly?

Reply:
I'm referring to Argon2 having deployed `$v=...$<params>`, versus `$v=...,<params>`